src/Reader/Reader.php

```diff
@@ -117,23 +118,30 @@ public static function setCache(CacheStorage $cache)
      *
      * Sets the HTTP client object to use for retrieving the feeds.
      *
-     * @param  ZendHttp\Client $httpClient
+     * @param  ZendHttp\Client | Http\ClientInterface $httpClient
      * @return void
      */
-    public static function setHttpClient(ZendHttp\Client $httpClient)
+    public static function setHttpClient($httpClient)
     {
+        if ($httpClient instanceof ZendHttp\Client) {
+            $httpClient = new Http\ZendHttpClientDecorator($httpClient);
+        }
+
+        if (! $httpClient instanceof Http\ClientInterface) {
```

Comment:
I was wrong here. HHVM tests is not failing. HHVM usually yields if you provide more arguments than needed to a method (extra arguments may require extra hw resources for nothing)